src/core/Note.dart

```diff
@@ -0,0 +1,128 @@
+import 'note_app.dart';
+class Note {
```

Comment:
Consider separating the note title and content into the note_model.dart file, and place note functions in a NoteManager class for better readability.